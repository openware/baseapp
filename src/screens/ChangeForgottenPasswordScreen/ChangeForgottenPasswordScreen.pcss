.pg-change-forgotten-password-screen {
  background: url(assets/images/background.svg);
  background-size: cover;
  display: flex;
  height: calc(100vh - (var(--big-gap) * 0.9834));
  justify-content: center;
  margin: 0 auto;

  .pg-change-forgotten-password-screen__container {
    background: transparent;
    margin-top: calc(var(--gap) * 3.34);
    width: calc(var(--gap) * 70.66);

    .cr-email-form {
      background: var(--body-background-color);
      min-height: calc(var(--big-gap) * 5.5);

      &__option {
        background: var(--subheader-background-color);
      }

      &__options-group {
        border-bottom: 2px solid var(--divider-color-level-2);
        color: var(--primary-text-color);
        height: calc(var(--big-gap) * 0.97);
        width: 100%;
      }

      &__label {
        background: var(--input-background-color);
        position: absolute;
        top: 0;
      }

      &__input {
        input {
          border: 1px solid var(--input-border-color);
          border-radius: calc(var(--gap) * 0.66);
          color: var(--input-text-color);
          font-size: calc(var(--gap) * 2.33);
        }
      }

      &__header {
        color: var(--secondary-contrast-cta-color);
        font-size: calc(var(--gap) * 2.33);
        font-style: normal;
        font-weight: normal;
        height: calc(var(--gap) * 4.33);
        line-height: calc(var(--gap) * 2.66);
        margin: calc(var(--gap) * 0.83);
        width: calc(var(--big-gap) * 6.3);
      }

      &__cros-icon {
        cursor: pointer;
        display: inline-block;
        height: calc(var(--gap) * 2.5);
        margin-top: calc(var(--gap) * 0.5);
        position: absolute;
        right: calc(var(--gap) * 3.3);
        width: calc(var(--gap) * 2.5);
      }

      &__button-wrapper {
        min-height: calc(var(--gap) * 10);
      }
    }

    .cr-email-form__option-inner {
      background: transparent;
      color: var(--primary-text-color);
      font-size: calc(var(--gap) * 2.33);
      font-style: normal;
      height: calc(var(--gap) * 9.66);
      line-height: calc(var(--gap) * 9.66);
      margin: 0 auto;
      text-align: center;
      text-transform: uppercase;
      width: 100%;
    }

    .cr-email-form__group--focused {
      .cr-email-form__label {
        color: var(--input-legend-color-active);
      }

      input {
        border: 1px solid var(--input-border-color-active) !important;
      }
    }
  }
}

@media screen and (max-width: 768px) {
  .pg-change-forgotten-password-screen {
    .pg-change-forgotten-password-screen__container {
      margin: calc(var(--gap) * 2.5) calc(var(--gap) * 1.25) 0;
    }
  }
}
