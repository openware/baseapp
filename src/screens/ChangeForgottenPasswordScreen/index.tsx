import * as React from 'react';
import { injectIntl } from 'react-intl';
import {
  connect,
  MapDispatchToPropsFunction,
  MapStateToProps,
} from 'react-redux';
import { RouterProps, withRouter } from 'react-router';
import { compose } from 'redux';
import { IntlProps } from '../../';
import { ChangePassword } from '../../components';
import { setDocumentTitle } from '../../helpers';
import {
    changeForgotPasswordFetch,
    changeLanguage,
    entropyPasswordFetch,
    RootState, selectChangeForgotPasswordSuccess,
    selectCurrentPasswordEntropy,
    selectMobileDeviceState,
} from '../../modules';

interface ChangeForgottenPasswordState {
    confirmToken: string;
}

interface ReduxProps {
    changeForgotPassword?: boolean;
    isMobileDevice: boolean;
    currentPasswordEntropy: number;
}

interface DispatchProps {
    changeForgotPasswordFetch: typeof changeForgotPasswordFetch;
    changeLanguage: typeof changeLanguage;
    fetchCurrentPasswordEntropy: typeof entropyPasswordFetch;
}

interface HistoryProps {
    history: {
        location: {
            search: string;
        };
    };
}

type Props = RouterProps & DispatchProps & HistoryProps & ReduxProps & IntlProps;

class ChangeForgottenPasswordComponent extends React.Component<Props, ChangeForgottenPasswordState> {
    constructor(props: Props) {
        super(props);

        this.state = {
            confirmToken: '',
        };
    }

    public componentDidMount() {
        setDocumentTitle('Change forgotten password');
        const { history } = this.props;
        const token = new URLSearchParams(history.location.search).get('reset_token');
        const lang = new URLSearchParams(history.location.search).get('lang');
        if (token) {
            this.setState({
                confirmToken: token,
            });
        }
        if (lang) {
            this.props.changeLanguage(lang.toLowerCase());
        }
    }

    public componentWillReceiveProps(next: Props) {
        if (next.changeForgotPassword && (!this.props.changeForgotPassword)) {
            this.props.history.push('/signin');
        }
    }

    public render() {
        const { isMobileDevice, currentPasswordEntropy } = this.props;

        return (
            <div className="pg-change-forgotten-password-screen">
                <div className="pg-change-forgotten-password-screen__container">
                    <ChangePassword
                        handleChangePassword={this.handleSendNewPassword}
                        title={!isMobileDevice && this.props.intl.formatMessage({id: 'page.header.signIn.resetPassword.title'})}
                        currentPasswordEntropy={currentPasswordEntropy}
                        fetchCurrentPasswordEntropy={this.props.fetchCurrentPasswordEntropy}
                        hideOldPassword={true}
                    />
                </div>
            </div>
        );
    }

    private handleSendNewPassword = payload => {
        const { confirmToken } = this.state;
        this.props.changeForgotPasswordFetch({
            ...payload,
            reset_password_token: confirmToken,
        });
    };
}

const mapStateToProps: MapStateToProps<ReduxProps, {}, RootState> = state => ({
    changeForgotPassword: selectChangeForgotPasswordSuccess(state),
    isMobileDevice: selectMobileDeviceState(state),
    currentPasswordEntropy: selectCurrentPasswordEntropy(state),
});

const mapDispatchToProps: MapDispatchToPropsFunction<DispatchProps, {}> =
    dispatch => ({
        changeForgotPasswordFetch: credentials => dispatch(changeForgotPasswordFetch(credentials)),
        changeLanguage: lang => dispatch(changeLanguage(lang)),
        fetchCurrentPasswordEntropy: payload => dispatch(entropyPasswordFetch(payload)),
    });

export const ChangeForgottenPasswordScreen = compose(
    injectIntl,
    withRouter,
    connect(mapStateToProps, mapDispatchToProps),
)(ChangeForgottenPasswordComponent) as React.ComponentClass;
