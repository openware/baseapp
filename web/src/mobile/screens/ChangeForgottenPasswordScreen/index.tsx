import * as React from 'react';
import { useIntl } from 'react-intl';
import { useHistory } from 'react-router';
import { ChangeForgottenPasswordScreen } from '../../../screens/ChangeForgottenPasswordScreen';
import { Modal } from '../../components';


const ChangeForgottenPasswordMobileScreen: React.FC = () => {
    const history = useHistory();
    const intl = useIntl();

    return <div>
        <Modal
            isOpen={true}
            onClose={() => history.push('/trading')}
            title={intl.formatMessage({ id: 'page.header.signIn.resetPassword.title' })}>
            <ChangeForgottenPasswordScreen/>
        </Modal>
    </div>;
};

export {
    ChangeForgottenPasswordMobileScreen,
};
