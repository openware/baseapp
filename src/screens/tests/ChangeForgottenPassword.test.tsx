import { shallow } from 'enzyme';
import React from 'react';
import { TestComponentWrapper } from 'lib/test';
import { IntlProps } from '../../';
import { ChangeForgottenPasswordScreen } from '../ChangeForgottenPasswordScreen';


const setup = (props: Partial<IntlProps> = {}) =>
    shallow(
        <TestComponentWrapper>
            <ChangeForgottenPasswordScreen />
        </TestComponentWrapper>
    );

describe('ChangeForgottenPasswordScreen test', () => {
    it('should render', () => {
        const wrapper = setup().render();
        expect(wrapper).toMatchSnapshot();
    });
});
