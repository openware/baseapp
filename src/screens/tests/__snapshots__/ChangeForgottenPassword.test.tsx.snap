// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangeForgottenPasswordScreen test should render 1`] = `ShallowWrapper {}`;
